### PR TITLE
feature/apple-musicKit-playing-error

### PR DESCRIPTION
**Pull request by JaHajRF**
apple music player error fixed

### CODE REVIEW

File: packages/webapp/src/containers/TurntableRoom.container.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ const pictures = [
   frontScreen,
 ]
 export const TurntableRoomContainer: React.FC = (): JSX.Element => {
-  const urlParams = useParams<{ roomName: 'audius' | 'pop' | 'apple' }>()
+  const urlParams = useParams<{ roomName: 'audius' | 'pop' | 'apple' | undefined }>()
```
Comment (kowalski) [inline packages/webapp/src/containers/TurntableRoom.container.tsx:78]: hmmm.... `undefined`?

Comment (kowalski) [reply packages/webapp/src/containers/TurntableRoom.container.tsx:78]: ok, nvm I see what you did with it

File: packages/webapp/src/store/appleMusicProvider/appleMusicProvider.saga.ts [hunk 2]
```diff
@@ -52,14 +54,35 @@ export function* initializeAppleMusicKit(dispatch: Dispatch<AnyAction>): Generat
...
+          ) {
+            dispatch(playerActions.songEnd(peerIdJson.id))
+          }
+        }
+        const forwardSongEvent = (eventOrError: { oldState: number; state: number }): void => {
+          if (
+            eventOrError.oldState === appleMusicPlaybackStates.stalled &&
+            eventOrError.state === appleMusicPlaybackStates.playing
+          ) {
+            const startTime = getState().player.startTime
```
Comment (kowalski) [inline packages/webapp/src/store/appleMusicProvider/appleMusicProvider.saga.ts:71]: slightly more solid way to use state:

```
const startTime = startTimeSelector(getState())
```

where startTimeSelector is use keySelector that you have

File: packages/webapp/src/store/index.types.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { AppleMusicProviderStore } from './appleMusicProvider/appleMusicProvider.reducer'
+import { ChatMessagesStore } from './chatMessages/chatMessages.reducer'
+import { MusicProviderTrackListStore } from './musicProvider/musicProvider.reducer'
+import { PlaylistStore } from './playlist/playlist.reducer'
+import { RouterStore } from './router/router.reducer'
+import { SessionStore } from './session/session.reducer'
+import { UserStore } from './user/user.reducer'
+import { WebPlayerStore } from './webPlayer/webPlayer.reducer'
+
+export type StoreType = {
```
Comment (kowalski) [inline packages/webapp/src/store/index.types.ts:18]: there already is something similar in `src/store/index.ts` :

```
interface CombinedReducer extends CommonCombinedReducer {
  appleMusicProvider: Reducer<AppleMusicProviderStore>
  audiusEndpoint: Reducer<GenericStore<string>>
  audiusTrackList: Reducer<GenericStore<Track[]>>
  chatMessages: Reducer<ChatMessagesStore>
  musicProviderEndpoint: Reducer<GenericStore<string>>
  musicProviderTrackList: Reducer<GenericStore<Track[]>>
  playlist: Reducer<PlaylistStore>
  router: Reducer<RouterStore>
  ui: Reducer<UiStore>
  user: Reducer<UserStore>
  webPlayer: Reducer<WebPlayerStore>
}
```

These approaches are okay, but there is a better way:

```
import { rootReducer } from './index.reducer'

const combinedReducer = combineReducers({
      ...rootReducer,
})
export type StoreType = ReturnType<typeof combinedReducer>
```

This way we only have 1 mapping to maintain (in index.reducer)

**Review by kowalski — APPROVED** [review]

**Comment by JaHajRF** [comment]
> you have `@ts-expect-error` because you didn't type `getState` correctly. Imo if you type it as returning `StoreType` the selector will work well.

well, its typed that way. Selector typing is the problem. It expects to get PlayerStore not entire Store.

**Comment by kowalski** [comment]
Ah, I see, weird. Ok ignore for now

**Pull request merged**